Tags resources as "Deleted" instead of deleting

tfrcarvalho opened the pull request:
TODO: update store test to include ListTagsForResource mock.

- `delete` command tags resources as delete instead of deleting (Goal: preventing accidental total loss of credentials to resources).
- Added force flag to actually delete the resource (previous default behavior)
- Updated `list`, `exec` and `read` commands to support new delete behavior.

tfrcarvalho committed:
Add support to mark as deleted.

tfrcarvalho committed:
Fix previous store tests.

tfrcarvalho committed:
Add test to store for resources tagged as deleted.

tfrcarvalho committed:
Merge branch 'master' into tiago/mark-deleted

tfrcarvalho committed:
Update README.

tzjames commented on README.md:
```diff
@@ -1,8 +1,8 @@
 # CZI - Chamber
 
 This repo is a fork of original Chamber tool ([segmentio/chamber](https://github.com/segmentio/chamber)). THer goal is to add the following functionality:
```
typo: THer -> The

tzjames commented on cmd/delete.go:
```diff
@@ -4,19 +4,24 @@ import (
+	// deleteCmd represents the delete command
+	deleteCmd = &cobra.Command{
+		Use:   "delete <service> <key>",
+		Short: "Tag a secreted as deleted (it will not be Delete a secret, including all versions",
```
This is a bit hard to read.  Am I getting at what you are trying to say with:
"Tag a secret as deleted. (Old versions of the secret will remain in the parameter store)

tzjames reviewed (changes requested):
Thanks Tiago!  Looks great.  Just a couple of small inline nits, plus I think you might not be handling an edge case.  If someone deletes a secret, but then tries and add it back in, ideally I think it should just remove the deleted tag and set the new value.

tfrcarvalho commented:
Yeah, I was going to add a "recover" command to explicitly recover the secret. I guess i can clear the tag when a new version is written.

tfrcarvalho committed:
Add ability to recover a parameter (recover command).
Remove deleted tag from parameter when a new value is written.
Fix delete command description.
Add option to include delete values when using list command.
Fix README.

tzjames commented on store/ssmstore.go:
```diff
@@ -115,6 +115,8 @@ func (s *SSMStore) Write(id SecretId, value string) error {
 		return err
 	}
 
```
I suppose this is mostly self explanatory, but you could add a comment.
// We remove any "deleted" tag in case the item was previously deleted.

tzjames reviewed (approved):
Updates look good!  Looking forward to start using this.
